diff --git a/content/child/child_thread_impl.cc b/content/child/child_thread_impl.cc
index 6a15d146b595f7462c74afb46f1f270b23554da3..6cf867347f511be895e4e58390f8f57f914c8ec5 100644
--- a/content/child/child_thread_impl.cc
+++ b/content/child/child_thread_impl.cc
@@ -184,6 +184,9 @@ class SuicideOnChannelErrorFilter : public IPC::MessageFilter {
     __lsan_do_leak_check();
 #endif
 #else
+#if defined(OS_MAC)
+    CleanupTor();
+#endif // defined(OS_MAC)
     base::Process::TerminateCurrentProcessImmediately(0);
 #endif
   }
