diff --git a/services/service_manager/sandbox/win/sandbox_win.cc b/services/service_manager/sandbox/win/sandbox_win.cc
index cde75bd07101098e5855adfc4d3bcf9c98faf3f5..962646de2f87060add50b323d40e96ff2d237677 100644
--- a/services/service_manager/sandbox/win/sandbox_win.cc
+++ b/services/service_manager/sandbox/win/sandbox_win.cc
@@ -892,6 +892,7 @@ sandbox::ResultCode SandboxWin::StartSandboxedProcess(
           service_manager::switches::kNoSandbox)) {
     base::LaunchOptions options;
     options.handles_to_inherit = handles_to_inherit;
+    BraveLaunchOption(process_type, &options);
     BOOL in_job = true;
     // Prior to Windows 8 nested jobs aren't possible.
     if (sandbox_type == SandboxType::kNetwork &&
