diff --git a/third_party/blink/renderer/modules/sanitizer_api/sanitizer.idl b/third_party/blink/renderer/modules/sanitizer_api/sanitizer.idl
index e8bf834ed90589f8fc0548fbb516471d6aed959a..55b1b38ca0eb92381860ec89e16bda2eef6a3806 100644
--- a/third_party/blink/renderer/modules/sanitizer_api/sanitizer.idl
+++ b/third_party/blink/renderer/modules/sanitizer_api/sanitizer.idl
@@ -8,9 +8,7 @@
   Exposed=Window,
   RuntimeEnabled=SanitizerAPI
 ] interface Sanitizer {
-  [RaisesException] constructor(optional SanitizerConfig config = {});
+  [RaisesException] constructor();
   [CallWith=ScriptState, RaisesException] DOMString sanitizeToString(DOMString input);
   [CallWith=ScriptState, RaisesException] DocumentFragment sanitize(DOMString input);
-
-  readonly attribute SanitizerConfig creationOptions;
 };
