diff --git a/third_party/blink/renderer/modules/sanitizer_api/sanitizer.cc b/third_party/blink/renderer/modules/sanitizer_api/sanitizer.cc
index d4db3e4658f2db94175f261308cb60e426708d16..93a3cba3e2cf277f6112cb152dffdd604424e315 100644
--- a/third_party/blink/renderer/modules/sanitizer_api/sanitizer.cc
+++ b/third_party/blink/renderer/modules/sanitizer_api/sanitizer.cc
@@ -4,7 +4,6 @@
 
 #include "sanitizer.h"
 
-#include "third_party/blink/renderer/bindings/modules/v8/v8_sanitizer_config.h"
 #include "third_party/blink/renderer/core/dom/document_fragment.h"
 #include "third_party/blink/renderer/core/editing/serializers/serialization.h"
 #include "third_party/blink/renderer/core/frame/local_dom_window.h"
@@ -14,13 +13,11 @@
 
 namespace blink {
 
-Sanitizer* Sanitizer::Create(const SanitizerConfig* config,
-                             ExceptionState& exception_state) {
-  return MakeGarbageCollected<Sanitizer>(config);
+Sanitizer* Sanitizer::Create(ExceptionState& exception_state) {
+  return MakeGarbageCollected<Sanitizer>();
 }
 
-Sanitizer::Sanitizer(const SanitizerConfig* config)
-    : config_(const_cast<SanitizerConfig*>(config)) {}
+Sanitizer::Sanitizer() = default;
 
 Sanitizer::~Sanitizer() = default;
 
@@ -47,13 +44,4 @@ DocumentFragment* Sanitizer::sanitize(ScriptState* script_state,
   return fragment;
 }
 
-SanitizerConfig* Sanitizer::creationOptions() const {
-  return config_;
-}
-
-void Sanitizer::Trace(Visitor* visitor) const {
-  ScriptWrappable::Trace(visitor);
-  visitor->Trace(config_);
-}
-
 }  // namespace blink
