diff --git a/third_party/blink/renderer/bindings/generated_in_modules.gni b/third_party/blink/renderer/bindings/generated_in_modules.gni
index dcae47d657da5f39bc16ebc903c9aa210d64fbc9..7c9a90496930865e1a8e3cdcd6415db593902b3b 100644
--- a/third_party/blink/renderer/bindings/generated_in_modules.gni
+++ b/third_party/blink/renderer/bindings/generated_in_modules.gni
@@ -637,8 +637,6 @@ generated_dictionary_sources_in_modules = [
   "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_rtc_session_description_init.h",
   "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_rtc_track_event_init.cc",
   "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_rtc_track_event_init.h",
-  "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_sanitizer_config.cc",
-  "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_sanitizer_config.h",
   "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_save_file_picker_options.cc",
   "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_save_file_picker_options.h",
   "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_scheduler_post_task_options.cc",
