diff --git a/third_party/blink/web_tests/external/wpt/interfaces/sanitizer-api.tentative.idl b/third_party/blink/web_tests/external/wpt/interfaces/sanitizer-api.tentative.idl
index 25cba22ecfc4ee194cf4f21aa2c0d24c7c16de05..433ad968a869b07407b14b7c726f2f194a8c6ecb 100644
--- a/third_party/blink/web_tests/external/wpt/interfaces/sanitizer-api.tentative.idl
+++ b/third_party/blink/web_tests/external/wpt/interfaces/sanitizer-api.tentative.idl
@@ -3,9 +3,6 @@
 [
   Exposed=Window
 ] interface Sanitizer {
-  [RaisesException] constructor(optional SanitizerConfig sanitizerConfig = {});
-  [CallWith=ScriptState, RaisesException] DocumentFragment sanitize(DOMString input);
-  [CallWith=ScriptState, RaisesException] DOMString sanitizeToString(DOMString input);
+  [RaisesException] constructor();
+  DOMString sanitizeToString(DOMString input);
 };
-
-dictionary SanitizerConfig {};
